feeds/wifi-ax/hostapd/patches/900-hapd-netlink-ubus-bridge.patch

```diff
@@ -0,0 +1,891 @@
++	int band;
++	uint32_t auth_status;
++};
++
++struct client_fdata_event {
++	unsigned char sta_mac[20];
++	uint64_t session_id;
++	struct timespec rx_ts;
++	struct timespec tx_ts;
++};
++
++struct client_ip_event {
++	unsigned char sta_mac[20];
++	uint64_t session_id;
++	unsigned char ip_addr[16];
++};
++
++struct client_session_record {
++	int type;
++	uint64_t timestamp;
++	union {
++		struct client_assoc_event assoc;
++		struct client_disassoc_event disassoc;
++		struct client_auth_event auth;
++		struct client_fdata_event fdata;
++		struct client_ip_event ip;
++	} u;
++};
++
++struct hostapd_event_avl_rec {
++	uint64_t session_id;
++	struct client_session_record *records;
++	size_t rec_nr;
++	struct avl_node avl;
+ };
+ 
+ void hostapd_ubus_add_iface(struct hostapd_iface *iface);
+@@ -45,6 +127,7 @@ void hostapd_ubus_add_bss(struct hostapd
+ void hostapd_ubus_free_bss(struct hostapd_data *hapd);
+ 
+ int hostapd_ubus_handle_event(struct hostapd_data *hapd, struct hostapd_ubus_request *req);
++int hostapd_ubus_handle_rt_event(struct hostapd_data *hapd, struct hostapd_ubus_request *req);
+ void hostapd_ubus_notify(struct hostapd_data *hapd, const char *type, const u8 *mac);
+ void hostapd_ubus_notify_beacon_report(struct hostapd_data *hapd,
+ 				       const u8 *addr, u8 token, u8 rep_mode,
+@@ -78,6 +161,11 @@ static inline int hostapd_ubus_handle_ev
+ {
+ 	return 0;
+ }
++
++static inline int hostapd_ubus_handle_rt_event(struct hostapd_data *hapd, struct hostapd_ubus_request *req)
++{
++	return 0;
++}
+ 
+ static inline void hostapd_ubus_notify(struct hostapd_data *hapd, const char *type, const u8 *mac)
+ {
+--- a/src/ap/hostapd.c
++++ b/src/ap/hostapd.c
+@@ -1406,19 +1406,22 @@ static int hostapd_setup_bss(struct host
+ 				   "Generic snooping infrastructure initialization failed");
+ 			return -1;
+ 		}
+-
+-		if (dhcp_snoop_init(hapd)) {
+-			wpa_printf(MSG_ERROR,
+-				   "DHCP snooping initialization failed");
+-			return -1;
+-		}
+-
+ 		if (ndisc_snoop_init(hapd)) {
+ 			wpa_printf(MSG_ERROR,
+ 				   "Neighbor Discovery snooping initialization failed");
+ 			return -1;
+ 		}
+ 	}
++	if (dhcp_snoop_init(hapd)) {
++		wpa_printf(MSG_ERROR,
++			   "DHCP snooping initialization failed");
++		hostapd_logger(
++			hapd, NULL, HOSTAPD_MODULE_IEEE80211,
++			HOSTAPD_LEVEL_DEBUG,
++			"dfranusic:DHCP snooping initialization failed");
++
++		return -1;
++	}
+ 
+ 	if (!hostapd_drv_none(hapd) && vlan_init(hapd)) {
+ 		wpa_printf(MSG_ERROR, "VLAN initialization failed.");
+--- a/src/ap/dhcp_snoop.c
++++ b/src/ap/dhcp_snoop.c
+@@ -40,6 +40,7 @@ static void handle_dhcp(void *ctx, const
+ 	int res, msgtype = 0, prefixlen = 32;
+ 	u32 subnet_mask = 0;
+ 	u16 ip_len;
++	int ubus_resp;
+ 
+ 	exten_len = len - ETH_HLEN - (sizeof(*b) - sizeof(b->exten));
+ 	if (exten_len < 4)
+@@ -112,6 +113,19 @@ static void handle_dhcp(void *ctx, const
+ 			   ipaddr_str(be_to_host32(b->your_ip)),
+ 			   prefixlen);
+ 
++			struct hostapd_ubus_request req = {
```

Comment:
Again, you said IP event was not working, is this change safe?

Reply:
Yes, it is safe, See the comment above